### PR TITLE
Update package format to julia 1.x

### PR DESCRIPTION
**Pull request by lucianolorenti**

### CODE REVIEW

File: Manifest.toml
Comment (omus) [inline Manifest.toml]: The Manifest.toml file probably shouldn't be included and be added to the .gitignore

File: Project.toml
Comment (omus) [inline Project.toml]: Most likely `Compat` was included to support Julia 1.0 when this package supported Julia 0.6. I would try removing this dependency and the `using Compat` calls and verify it is still needed.

File: Project.toml
Comment (omus) [inline Project.toml:9]: Double check for conditional code based upon the version like: https://github.com/ssfrr/DeepDiffs.jl/blob/cb5550b5e2d002285bc2341438d02afa045e37a7/src/DeepDiffs.jl#L17 as these can be removed

Comment (omus) [reply Project.toml:9]: The `.travis.yml` and `appveyor.yml` need to be updated to no longer test against Julia 0.6

File: appveyor.yml
Comment (omus) [inline appveyor.yml]: Looks like this file hasn't been updated in some time. There is now an easier way: https://github.com/JuliaCI/Appveyor.jl

This should be the updated form I think:

```yaml
environment:
  matrix:
  - julia_version: 1
  - julia_version: nightly

platform:
  - x86 # 32-bit
  - x64 # 64-bit

branches:
  only:
    - master
    - /release-.*/

notifications:
  - provider: Email
    on_build_success: false
    on_build_failure: false
    on_build_status_changed: false

install:
  - ps: iex ((new-object net.webclient).DownloadString("https://raw.githubusercontent.com/JuliaCI/Appveyor.jl/version-1/bin/install.ps1"))

build_script:
  - echo "%JL_BUILD_SCRIPT%"
  - C:\julia\bin\julia -e "%JL_BUILD_SCRIPT%"

test_script:
  - echo "%JL_TEST_SCRIPT%"
  - C:\julia\bin\julia -e "%JL_TEST_SCRIPT%"
```

File: .travis.yml
Comment (omus) [inline .travis.yml:8]: Will automatically update to the latest verison 1 release
```suggestion
  - 1
```

File: Manifest.toml
Comment (omus) [reply Manifest.toml]: Addressed

File: Project.toml
Comment (omus) [reply Project.toml]: Addressed

File: Project.toml
Comment (omus) [reply Project.toml:9]: Addressed

File: appveyor.yml
Comment (lucianolorenti) [reply appveyor.yml]: It worked!

**Review by omus — APPROVED** [review]

**Comment by omus** [comment]
cc: @ssfrr

File: Project.toml
Comment (johnnychen94) [inline Project.toml]:
```suggestion
version = "1.2.0"
```

**Comment by johnnychen94** [comment]
This package locks the `Compat` version to v"2" in [ReferenceTests](https://github.com/Evizero/ReferenceTests.jl), which turns out to cause the issue https://github.com/zygmuntszpak/ImageBinarization.jl/pull/58 so I'm wondering if we can get some updates here since it looks like `Compat` is no longer needed here.

Edit: the issue got solved in ImageBinarization side

File: .travis.yml
Comment (johnnychen94) [reply .travis.yml:8]: I think it would be helpful to add 1.0 tests for both travis and appveyor.

**Comment by ssfrr** [comment]
I made a couple minor tweaks. Thanks all for the PR and reviews. I'll get a release out shortly

**Comment by omus** [comment]
Thanks for getting this merged. If you want assistance with this package I'd be happy to help out as a maintainer.

**Comment by ssfrr** [comment]
Release tagged, just waiting for it to be merged into General:

https://github.com/JuliaRegistries/General/pull/11174

Thanks for the offer of help! I'd be happy and grateful to add you as a maintainer.

**Pull request merged**
